etlman/projects/views.py

```diff
@@ -76,3 +75,135 @@ def new_project_view(request):
     form = ProjectForm()
     context = {"form": form, "username": username}
     return render(request, "projects/new_project.html", context)
+
+
+@authorize(user_is_project_collaborator)
+def pipeline_list(request, project_id):
+    project = get_object_or_404(Project, pk=project_id)
+    pipelines = Pipeline.objects.filter(project=project)
+    context = {"pipeline_list": pipelines, "current_project": project}
+    return render(request, "projects/pipeline_list.html", context)
+
+
+@authorize(user_is_project_collaborator)
+def new_pipeline_step1(request, project_id):
+    project = get_object_or_404(Project, pk=project_id)
+    username = request.user.username
+    # Form functionality
+    if request.method == "POST":
+        form_pipeline = PipelineForm(request.POST)
+        form_datainterface = DataInterfaceForm(request.POST)
+        if form_pipeline.is_valid() and form_datainterface.is_valid():
+
+            filled_datainterface = form_datainterface.save(commit=False)
+            filled_datainterface.name = request.POST.getlist("name")[-1]
+            request.session["data_interface"] = model_to_dict(filled_datainterface)
+
+            filled_pipeline = form_pipeline.save(commit=False)
+            filled_pipeline.name = request.POST.getlist("name")[0]
+            filled_pipeline.input_id = filled_datainterface.pk
+            request.session["pipeline"] = model_to_dict(filled_pipeline)
+
+            return HttpResponseRedirect(
+                reverse("projects:new_step", args=(project.pk,))
+            )
+    else:  # GET
+        # If "in_transaction" is on session, the user came back from multiform step2
```

Comment:
Right now if you refresh page one after going back from page 2, it clears the values of the form. It would be nice to persist that here. The more I think about it, it might not even be necessary to clear the values in case someone comes back later. Some web apps (github for example) will save the contents of this text field even if I close it and come back.

Reply:
Done!